scripts/ccpp_state_machine.py

```diff
@@ -3,11 +3,11 @@
 # CCPP framework imports
 from state_machine import StateMachine
 
-_INIT_ST = r"(?:(?i)init(?:ial(?:ize)?)?)"
-_FINAL_ST = r"(?:(?i)final(?:ize)?)"
-_RUN_ST = r"(?:(?i)run)"
-_TS_INIT_ST = r"(?:(?i)timestep_init(?:ial(?:ize)?)?)"
-_TS_FINAL_ST = r"(?:(?i)timestep_final(?:ize)?)"
+_INIT_ST = r"(?:init(?:ial(?:ize)?)?)"
+_FINAL_ST = r"(?:final(?:ize)?)"
+_RUN_ST = r"(?:run)"
+_TS_INIT_ST = r"(?:timestep_init(?:ial(?:ize)?)?)"
+_TS_FINAL_ST = r"(?:timestep_final(?:ize)?)"
```

Comment:
What is the thinking behind this change?
Is it the re.IGNORECASE usage elsewhere?

Reply:
@gold2718 
FWIW. Looks like this was a change from main, https://github.com/NCAR/ccpp-framework/commit/24bab13a6e322bf528a184f4261bf356dfbd7b66

Reply:
Yes, back in the good old days I fixed this code that's no longer allowed in Python 3.11+